### DISCUSSION TITLE
datetime format YYYYMMDD (Int32) - quickfilter

### DISCUSSION THREAD

**Discussion by tmartinscopo** [serenity-is/Serenity#6529] [Help] [answered]
Hi,

In my database dates fields are of type int32 (YYYYMMDD).
How can we make date quickfilters work well with this data type in dates? Any idea?

![image](https://user-images.githubusercontent.com/37545074/189119929-62f50dd5-f4d3-4d1c-b49b-f0d3b0b32358.png)

Thx

**Comment by edwardch** [comment, marked as answer]
Any reason to use INTs for dates? If you use the SQL Date type it's 3 bytes, vs INT which is 4 bytes.

If you need to do int you might update your getQuickFilters() method in your grid.ts file.

Something like this, I guess
```ts
filters.push({
    type: Serenity.DateEditor,
    field: 'Something',
    title: 'Date',
    handler: w => {
        var intDate = w.value;
        // var intValue = your code to convert intDate to an int;
        // add code to send through equality on your int field
        // (this.view.params as Serenity.ListRequest).EqualityFilter
    }
})
```

**Comment by tmartinscopo** [comment]
92 / 5 000
@edwardch the information is imported from another type of database and was not converted to SQL Date type.
Thta's one way to do it, Thanks
